runtime/jvmti/jvmtiMemory.c

```diff
@@ -80,4 +80,21 @@ jvmtiDeallocate(jvmtiEnv* env,
 	TRACE_JVMTI_RETURN(jvmtiDeallocate);
 }
 
-
+jvmtiError JNICALL 
+jvmtiSetHeapSamplingInterval(jvmtiEnv *env, 
+	jint samplingInterval)
+{
+	jvmtiError rc = JVMTI_ERROR_NONE;
+	J9VMThread *currentThread = NULL;
+	
+	Trc_JVMTI_jvmtiSetHeapSamplingInterval_Entry(env, samplingInterval);
+	
+	rc = getCurrentVMThread(((J9JVMTIEnv *)env)->vm, &currentThread);
+	if ((JVMTI_ERROR_NONE == rc) && (NULL != currentThread)) {
+		J9MemoryManagerFunctions *fns = currentThread->javaVM->memoryManagerFunctions;
+		if (NULL != fns) {
+			fns->j9gc_set_allocation_sampling_interval(currentThread, samplingInterval);
```

Comment:
what happens if we fail (eg. pass in a sampling interval of 1TB).
  
should this return an internal error? or fail silently

Reply:
Any value larger than `Integer.MAX_VALUE` will be casted to `jint`, and the negative value is going to be caught by `ENSURE_NON_NEGATIVE(samplingInterval)`.